### ISSUE TITLE
[Feature request] Image matrix generation without all permutations

### ISSUE THREAD

**Issue by nahojjjen** [hlky/stable-diffusion-webui#118]
The current prompt matrix is very helpful, but Ive found that for my use-cases "all permutations" is rarely  what I want, Id rather create a matrix like this:

![bild](https://user-images.githubusercontent.com/8545451/187028425-09bf4603-76a1-4fb2-9b8b-c9ffcf74ad7b.png)

With the current matrix feature, Id need to do three different batches, which would also generate several unnecessary images:
a dog | by studio ghibli | digital art, vector | photography. bokeh
an eagle | by studio ghibli | digital art, vector | photography. bokeh
an elephant| by studio ghibli | digital art, vector | photography. bokeh

**Comment by hlky** [comment]
Is this what you want #131 #134 ?

**Comment by nahojjjen** [comment]
@hlky Yes, this was what I wanted :) 
Thanks hlky, and thanks @JustOxlamon for implementing it.

I initially didn't get it working because I failed to see that the @ symbol was needed, I realized when I read through your code changes. 

Ive got it working now, Ive got a minor issue when there's a bigger grid than 2*2, where the resulting bottom half of the result has filled with pure black. At least on my machine. Have I broken something, or can someone else reproduce? 

Example prompt to reproduce:
"@ ( green | red | blue ) ( ball | rock | house) "

results in a single-column image with all 9 images:

![grid-00192-3836482269_blue_house](https://user-images.githubusercontent.com/8545451/187047750-7369ea35-2b25-45f4-afec-8cddf13d45db.jpg)


But overall the feature works great! :) @hlky Are above mentioned points something you want me to create a new github-issue for?
